pkg/virt-controller/watch/vm.go

```diff
@@ -1532,6 +1555,35 @@ func (c *VMController) deleteOlderVMRevision(vm *virtv1.VirtualMachine) (bool, e
 	return createNotNeeded, nil
 }
 
+func (c *VMController) deleteVMRevisions(vm *virtv1.VirtualMachine) error {
+	keys, err := c.crInformer.GetIndexer().IndexKeys("vm", string(vm.UID))
+	if err != nil {
+		return err
+	}
+
+	for _, key := range keys {
+		if !strings.Contains(key, vmRevisionName(vm.UID)) {
+			continue
+		}
+
+		storeObj, exists, err := c.crInformer.GetStore().GetByKey(key)
+		if !exists || err != nil {
+			return err
+		}
+		cr, ok := storeObj.(*appsv1.ControllerRevision)
+		if !ok {
+			return fmt.Errorf("unexpected resource %+v", storeObj)
+		}
+
+		err = c.clientset.AppsV1().ControllerRevisions(vm.Namespace).Delete(context.Background(), cr.Name, v1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
for future improvements: add a custom label, such as `kubevirt.io/vm-name: <vm-name>`, to the controller revision resource creation.
Than here we can simply do:
```
		err = c.clientset.AppsV1().ControllerRevisions(vm.Namespace).DeleteCollection(context.Background(), v1.DeleteOptions{}, v1.ListOptions{LabelSelector: fmt.Sprintf("kubevirt.io/vm-name==%s", vm.Name)})
		if err != nil {
			return err
		}
```